### PR TITLE
Add Text StartsWith And EndsWith Predicates

### PR DESCRIPTION
**Pull request by haspadar**
- Added Text\StartsWith and Text\EndsWith as Scalar<bool> predicates delegating to PHP str_starts_with and str_ends_with
- Tests cover empty needle, exact match, needle longer than haystack and UTF-8 multibyte input
- Switched the PHPStan afferent-coupling rule to ignoreAbstract, retiring the per-class inline suppression on TextEnvelope and the ScalarEnvelope entry from excludedClasses

## Release Notes

* **New Features**
  * Introduced `StartsWith` utility to check if text begins with a specified substring
  * Introduced `EndsWith` utility to check if text ends with a specified substring
  * Both utilities include comprehensive UTF-8 multibyte character support for global text operations

* **Tests**
  * Added test coverage for new text utilities, including edge cases and UTF-8 character handling

[Review Change Stack](https://app.coderabbit.ai/change-stack/haspadar/primus/pull/207?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/Text/EndsWith.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+final readonly class EndsWith extends ScalarEnvelope
+{
+    /**
+     * Ctor.
+     *
+     * @param Text $haystack The text to inspect.
+     * @param Text $needle The suffix to look for.
+     */
+    public function __construct(Text $haystack, Text $needle)
```
[the remark this answers is not in the document]
Comment (haspadar) [reply src/Text/EndsWith.php:31]: Skipping. `StartsWith::__construct(Text, Text)` and `EndsWith::__construct(Text, Text)` change the signature compared to `ScalarEnvelope::__construct(Scalar $origin)`, so they are new constructors rather than overrides. PHP `#[Override]` is for true replacements of an inherited method; applying it to a different signature would be misleading. Existing Primus decorators (`IsEmpty`, `Constant`, etc.) follow the same convention.

File: src/Text/StartsWith.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+final readonly class StartsWith extends ScalarEnvelope
+{
+    /**
+     * Ctor.
+     *
+     * @param Text $haystack The text to inspect.
+     * @param Text $needle The prefix to look for.
+     */
+    public function __construct(Text $haystack, Text $needle)
```
[the remark this answers is not in the document]
Comment (haspadar) [reply src/Text/StartsWith.php:31]: Same reason as the EndsWith reply: `StartsWith::__construct(Text, Text)` is a new constructor, not an override of `ScalarEnvelope::__construct(Scalar $origin)`. `#[Override]` would be misleading; existing decorators (`IsEmpty`, etc.) follow the same convention.
